internal/commands/pull/inputs_test.go

```diff
@@ -159,3 +160,169 @@ func TestPullInputsResolveRemoteApp(t *testing.T) {
 		assert.Equal(t, errors.New("something bad happened"), err)
 	})
 }
+
+func TestPullTemplatesResolve(t *testing.T) {
+	templateZipPkg1, err := zip.OpenReader("testdata/template_1.zip")
+	assert.Nil(t, err)
+	defer templateZipPkg1.Close()
+
+	templateZipPkg2, err := zip.OpenReader("testdata/template_2.zip")
+	assert.Nil(t, err)
+	defer templateZipPkg2.Close()
+
+	t.Run("should not do anything if fetching for compatible templates errors", func(t *testing.T) {
+		var realmClient mock.RealmClient
+
+		realmClient.CompatibleTemplatesFn = func(groupID, appID string) ([]realm.Template, bool, error) {
+			return nil, false, errors.New("something went wrong")
+		}
+
+		_, ui := mock.NewUI()
+		input := inputs{}
+		templateZipMap, err := input.resolveClient(ui, realmClient, "some-group-id", "some-app-id")
+		assert.Nil(t, templateZipMap)
+		assert.Equal(t, "something went wrong", err.Error())
+	})
+
+	t.Run("with a template ID passed in", func(t *testing.T) {
+		t.Run("should return an error if fetching the ClientTemplate for this template fails", func(t *testing.T) {
```

Comment:
in addition to avoiding capitalization/punctuation, avoid "code words" like variable or function names in the test descriptions as well

```suggestion
		t.Run("should return an error if fetching the client template for this template fails", func(t *testing.T) {
```